test/test_demos.py

```diff
@@ -0,0 +1,41 @@
+"""
+Checks that all demo scripts run.
+"""
+import glob
+import os
+import pytest
+import shutil
+import subprocess
+import sys
+
+
+# Set environment variable to specify that tests are being run so that we can
+# cut down the length of the demos
+os.environ["ANIMATE_REGRESSION_TEST"] = "1"
+
+cwd = os.path.abspath(os.path.dirname(__file__))
+demo_dir = os.path.abspath(os.path.join(cwd, "..", "demos"))
+all_demos = glob.glob(os.path.join(demo_dir, "*.py"))
+
+
+@pytest.fixture(params=all_demos, ids=lambda x: os.path.basename(x))
```

Comment:
The way things are set up here seems to want to future proof for a situation where we have multiple subdirectories under demos/ but in that case I foresee you would not necessarily have unique, meaningful (base) filename (this is currently an issue in Thetis)

Reply:
To be honest, this was copied from Goalie, which was probably copied from Thetis at some point! Tweaked in https://github.com/pyroteus/animate/pull/63/commits/95b9f3257e2766cfb5aca5d275ceb6af31ab72f5.